src/Reflex/Class.hs

```diff
@@ -791,6 +795,64 @@ zipListWithEvent f l e = do
   mapAccumMaybe_ f' l e
 
 --------------------------------------------------------------------------------
+-- AttachApply
+--------------------------------------------------------------------------------
+
+-- | An 'AttachApply' type carries a function that can be applied to an 'Event' whenever the 'Event' fires.
+--
+-- 'AttachApply' is intended to be used via the '<@>' and '<@' operators and in conjunction with the `Applicative` interface.
+--
+-- If we have:
+-- > f  :: a -> b -> c -> d
+-- > g  :: a -> b -> d
+-- > b1 :: Behavior t a
+-- > b2 :: Behavior t b
+-- > e  :: Event t c
+-- and an 'AttachApply' instance for `Behavior`, then we get:
+-- > f <$> b1 <*> b2 <@> e :: Event t d
+-- > g <$> b1 <*> b2 <@  e :: Event t d
+--
+-- It is a handy way to combine several 'Behavior's or 'Dynamic's together so that you can sample them all at the times when an `Event` is firing.
+class Reflex t => AttachApply t f where
+  attachApply :: f (a -> b) -> Event t a -> Event t b
+
+instance Reflex t => AttachApply t Identity where
+  attachApply (Identity f) = fmap f
+
+instance Reflex t => AttachApply t Maybe where
+  attachApply (Just f) e = fmap f e
+  attachApply Nothing  _ = never
+
+instance Reflex t => AttachApply t (Behavior t) where
+  attachApply = attachWith ($)
+
+instance Reflex t => AttachApply t (Dynamic t) where
+  -- This is an inlined version of attachPromptyDynWith.
+  -- It is inlined to avoid having an orphan instance.
```

Comment:
That's a very good point, but I think we should just move that function here, rather than repeating its body.

Also, I think it's worth considering whether the default operation ought to be attachPromptlyDynWith or a non-prompt version.  In the vast majority of code, I find that the prompt version is not necessary, and it make code more prone to loops.